ratings/rate_subdir.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import sys
+sys.path.insert(0, '.')
+
+from absl import flags
+from absl import app
+
+import choix
+import fnmatch
+import numpy as np
+import os
+import re
+import random
+import subprocess
+import math
+from tqdm import tqdm
+import datetime as dt
+from collections import defaultdict
+from ratings import math_ratings
+
+
+FLAGS = flags.FLAGS
+
+PW_REGEX = "PW\[([^]]*)\]"
+PB_REGEX = "PB\[([^]]*)\]"
+RESULT_REGEX = "RE\[([^]]*)\]"
+
+
+def extract_pairwise(files):
+    ids = defaultdict(lambda: len(ids))
+    results = []
+    for _file in tqdm(files):
+        with open(_file) as f:
+            text = f.read()
+        pw = re.search(PW_REGEX, text)
+        pb = re.search(PB_REGEX, text)
+        result = re.search(RESULT_REGEX, text)
+        if not (pw and pb and result):
+            print("Player or result fields not found: ", _file)
+            continue
+
+        pw = pw.group(1)
+        pb = pb.group(1)
+        result = result.group(1)
+
+        if pw == pb:
+            print("Players were the same: ", _file)
+            continue
+        if result.lower().startswith('b'):
+            results.append( (ids[pb], ids[pw]) )
+        if result.lower().startswith('w'):
+            results.append( (ids[pw], ids[pb]) )
+    
+    return ids, results
+
+
+def fancyprint_ratings(ids, ratings, results=None):
+    player_lookup = {v:k for k,v in ids.items()}
+    HEADER = "\n{:<25s}{:>8s}{:>8s}{:>8}{:>7}-{:<8}" 
+    ROW = "{:<25.23s} {:6.0f}  {:6.0f}  {:>6d}  {:>6d}-{:<6d}"
+
+    if not results:
+        for pid, (rating, sigma) in sorted(ratings.items(), 
```

Comment:
maybe breakout the sorted(ratings....) as sorted_player_ratings = ...

Reply:
Done.